pr-type/feature-development: Jira Sprint Board (Grafana Dashboard)

Bhoopalan1999 opened the pull request:
Adds a new Grafana dashboard for Jira that is sprint-based rather than time-based. Some of the features:

Hides the global time picker
Introduces a sprint selection dropdown to scope all panels Does not modify the existing Kanban (time-based) dashboard

<!--
Licensed to the Apache Software Foundation (ASF) under one or more
contributor license agreements.  See the NOTICE file distributed with
this work for additional information regarding copyright ownership.
The ASF licenses this file to You under the Apache License, Version 2.0
(the "License"); you may not use this file except in compliance with
the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
-->
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [ ] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [ ] I have added relevant tests.
- [ ] I have added relevant documentation.
- [ ] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

<!--
Thanks for submitting a pull request!

We appreciate you spending the time to work on these changes.
Please fill out as many sections below as possible.
-->

### Summary
What does this PR do?
The PR adds a new Grafana dashboard for Jira. While the existing Jira dashboard is used for time based Kanban boards, this PR contributes an additional Jira board which is used for sprint boards. The board doesn't have time picker. Instead, I have added sprint dropdown. All tiles correspond to the sprint dropdown. 

### Screenshots
**Sprint dropdown with 'Multiple selection' and 'All' option. Time picker is removed**
<img width="958" height="46" alt="image" src="https://github.com/user-attachments/assets/1d965ff5-83eb-4f89-a402-c78eaf684aa8" />
**Entire Dashboard screenshot**
<img width="941" height="356" alt="image" src="https://github.com/user-attachments/assets/85039e58-3c15-459d-b5c3-079e556b4235" />

Bhoopalan1999 commented:
Hello @Startrekzky 

Thanks for the feedback. 

**1. Difference between Cycle Lead Time & Issue Lead Time:**
**Cycle time** calculates how long has it taken for an issue to be resolved from the time it was moved to 'In Progress' or 'Developing'. That's why you see only these two Jira status selected in the sql query. 
**Issue lead time**, on the other hand, calculates the entire lifetime of a specific Jira issue (from the time of the issue creation -> till it is marked as 'delivered'). Considering this is a sprint board, I thought having both would be useful for sprint board users as most of my teams who opted for sprint board needed Cycle time. I shall remove it and update you so that it is consistent with existing Jira Kanban board. 

**2. The number of overall issue delivery rate is not consistent with the rate by sprint**
It's working in my dashboards. I'll troubleshoot and fix it and raise another PR. 

**3. Please remove all status/type/... filters only applicable to your own use cases.**
Sure. I'll remove that and update in the next PR.

Startrekzky commented on grafana/JiraSprintBoard.json:
```diff
@@ -0,0 +1,1058 @@
+      "type": "row"
+    },
+    {
+      "datasource": {
```
You should replace all occurences of
```
      "datasource": {
        "type": "mysql",
        "uid": "cey4ufm78rw8wf"
      },
```
with
```
"datasource": "mysql",
```

Otherwise, the dashboard can not be displayed properly.

Bhoopalan1999 commented on grafana/JiraSprintBoard.json:
```diff
@@ -0,0 +1,1058 @@
+    "to": "now"
+  },
+  "timepicker": {
+    "hidden": true
```
Yes, A time filter is not used in Sprint board. Hence removed it

Bhoopalan1999 commented:
Hello @Startrekzky 

Thanks for the feedback. I've updated all the changes requested. Also, I've now moved the file. Previously, I've accidentally kept the file inside grafana folder directly instead of grafana/dashboards. That's fixed now. Please review.
